js/apps/system/_admin/aardvark/APP/react/src/views/analyzers/forms/TextForm.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { ChangeEvent, useEffect } from "react";
+import { FormProps } from "../constants";
+import CaseInput from "./CaseInput";
+import { filter, get, has, isEmpty, negate } from 'lodash';
+
+const EdgeNGramInput = ({ formState, updateFormField }: FormProps) => {
+  const updateMinLength = (event: ChangeEvent<HTMLInputElement>) => {
+    updateFormField('properties.edgeNgram.min', parseInt(event.target.value));
+  };
+
+  const updateMaxLength = (event: ChangeEvent<HTMLInputElement>) => {
+    updateFormField('properties.edgeNgram.max', parseInt(event.target.value));
+  };
+
+  const togglePreserve = () => {
+    updateFormField('properties.edgeNgram.preserveOriginal', !get(formState, ['properties', 'edgeNgram', 'preserveOriginal']));
+  };
+
+  return <fieldset>
+    <legend style={{
```

Comment:
Is there maybe a way to re-use those stylings in different modals we will use in the future as well? 
(would be better instead of re-defining it every time. )

Reply:
Yes agreed, these styled components should be re-packaged for better portability. Will work on this in this PR.

Reply:
Done.